### ISSUE TITLE
quarkus-rest-client-reactive-jackson doesn't provide capabilities

### ISSUE THREAD

**Issue by chberger** [quarkusio/quarkus#31756]
### Description

Currently, I'm supporting the [quarkus-openapi-generator](https://github.com/quarkiverse/quarkus-openapi-generator) team to add support for ``RESTEasy Reactive``. During the implementation we've figured out that ``quarkus-rest-client-reactive-jackson`` doesn't provide any capabilities while ``quarkus-rest-client-jackson`` does. 

However, checking for ``rest-client-reactive-jackson`` capabilities would be crucial for us to identify which version (classic vs. reactive) is being used by the business application and whether the expected ``Jackson`` dependency is present as well, which is a prerequisite for our generator.

While the check for ``quarkus-rest-client-jackson``  is straightforward:

```
 private boolean isJacksonClassicClientPresent(CodeGenContext context) {
        return context.applicationModel().getExtensionCapabilities().stream()
                .flatMap(extensionCapability -> extensionCapability.getProvidesCapabilities().stream())
                .anyMatch(Capability.RESTEASY_JSON_JACKSON_CLIENT::equals);
    }
```

We had to implement a workaround for the ``quarkus-rest-client-reactive-jackson`` check:

```
private boolean isJacksonReactiveClientPresent(CodeGenContext context) {
    return context.applicationModel().getDependencies().stream()
            .anyMatch(this::isJacksonReactiveClient);
}

private boolean isJacksonReactiveClient(ResolvedDependency resolvedDependency) {
      return "quarkus-rest-client-reactive-jackson".equals(resolvedDependency.getArtifactId())
              && "io.quarkus".equals(resolvedDependency.getGroupId());
  }

```
Now we're wondering why ``ResolvedDependency`` comes from a ``Maven Quarkus`` package and whether this code would work in case a business app would use ``Gradle``, but not ``Maven``.

Basically it would be very helpful when ``quarkus-rest-client-reactive-jackson`` would provide capabilities, cause then we could continue with the same approach as with ``quarkus-rest-client-jackson``.

Please refer to this [PR 262](https://github.com/quarkiverse/quarkus-openapi-generator/pull/262) if you need further details.

Thanks in advance

### Implementation ideas

Extend ``quarkus-extension.yaml`` of ``quarkus-rest-client-reactive-jackson`` with the requested capabilities like:

```
  capabilities:
    provides:
    - "io.quarkus.rest.client.reactive.jackson"
```

Define a constant like this Capability.RESTEASY_JSON_JACKSON_REACTIVE_CLIENT

**Comment by geoand** [comment]
@chberger Thanks for reporting.

Would you like to contribute this fix?

**Comment by chberger** [comment]
@geoand I'd love to do that. I'll open a PR once it's implemented. 
Anyway, would be cool to get an answer to my question regarding the ``ResolvedDependency`` and the ``Quarkus Maven`` relation ...

**Comment by geoand** [comment]
> I'd love to do that. I'll open a PR once it's implemented.

💪🏼 

> Anyway, would be cool to get an answer to my question regarding the ResolvedDependency and the Quarkus Maven relation

That's one for @aloubyansky

**Comment by aloubyansky** [comment]
Yes, it will work in Gradle, of course. Practically, all the artifacts are coming from Maven repositories, that's the reason the API is following Maven artifact conventions.

**Comment by chberger** [comment]
All right, makes sense! Thanks for the explanation ..

**Comment by chberger** [comment]
Hmm still struggling with a proper capability name. Don't understand the naming convention ... :-(

Status quo:

| Extension name| Capability name |
|---------|------|
| quarkus-rest-client | io.quarkus.rest.client  |
| quarkus-rest-client-jackson | io.quarkus.resteasy.json.jackson.client |
| quarkus-rest-client-reactive | io.quarkus.rest.client.reactive |

So for ``quarkus-rest-client-reactive-jackson`` it's either ``io.quarkus.rest.client.reactive.jackson`` or ``io.quarkus.resteasy.json.jackson.client.reactive``.

@geoand wdyt?

**Comment by chberger** [comment]
or even ``io.quarkus.resteasy.reactive.json.jackson.client``

**Comment by geoand** [comment]
I propose `io.quarkus.rest.client.reactive.jackson`
